src/server/pps/server/pipeline_controller.go

```diff
@@ -359,9 +359,10 @@ func (op *pipelineOp) rcIsFresh() bool {
 		log.Errorf("PPS master: auth token in %q is stale %s != %s",
 			op.ptr.Pipeline.Name, rcAuthTokenHash, hashAuthToken(op.ptr.AuthToken))
 		return false
-	case rcSpecCommit != op.ptr.SpecCommit.ID:
+	// TODO(global ids): this won't trigger properly on start/stop pipeline which no longer updates the pipeline's OriginalSpecCommit
```

Comment:
Still reading through api_server.go, but does `CreatePipeline` update the pipeline version? It may be all that's needed to fix this is to change the tag in `worker_rc.go`  from `specCommitAnnotation` to `pipelineVersionAnnotation` and then compare that here

Reply:
Yes, but the thing I am worried about is start and stop pipeline, since those would create new spec commits (although I'm not entirely sure that's necessary) - I'm working on fixing those tests right now in the `grey/global-ids-pps-tests` branch.